### ISSUE TITLE
Failing the last test

### ISSUE THREAD

**Issue by Odog1996** [HPCE/hpce-2017-cw3#97]
I'm failing test 27 which tests V4 against V5 and expects V5 to be 1.01x faster. Do we need to have implemented another optimisation beyond what is done in the walkthrough in order to pass this test?

**Comment by swarth100** [comment]
Additionally would like to point out the world size is way too small to get a speedup v4 -> v5.

Speedups to occur when I test for big worlds (incredibly similarly to what is outlined in the spec), though no speedup happens with the auto-testing testcase.

V4 and v5 randomly take either `0.34` or `0.35`s

If I get v4 on `0.35`s and v5 on `0.34` I pass the test.

Should I add code that slows down v4 on purpose? (Which is not optimised out by -O3?)

**Comment by jjd06** [comment]
Yes, this test is somewhat statistical -- you are basically playing in the noise. You will have to do a bit more work to make it pass reliably. I've seen that some students have done this. A clever (and cheeky) solution is, indeed, to slow v4 down (a little bit), but that isn't really in keeping with the spirit of things! As far as the assessment is concerned, it doesn't make much of a difference: you're going to get 96% if you pass 27 of the 28 tests.
